populator-machinery/controller_test.go

```diff
@@ -349,7 +391,23 @@ func runSyncPvcTests(tests []testCase, t *testing.T) {
 			}
 			err := compareNotifyMap(test.expectedKeys, c.notifyMap)
 			if err != nil {
-				t.Errorf(err.Error())
+				t.Errorf("Compare notifyMap failed, error: %+v", err.Error())
+			}
+			actualPvcPrime, err := c.kubeClient.CoreV1().PersistentVolumeClaims(testVpWorkingNamespace).Get(context.TODO(), testPvcPrimeName, metav1.GetOptions{})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					t.Errorf("Get PVC prime failed, error: %+v", err.Error())
+				}
+			}
+			var actualMutateObjects mutateObjects
+			if actualPvcPrime != nil {
+				actualMutateObjects = mutateObjects{
+					pvcPrime: actualPvcPrime,
+				}
+			}
+			err = compareMutateObjects(&test.expectedMutateObjects, &actualMutateObjects)
```

Comment:
This naming is a little inconsistent. In the cases where the test didn't actually test mutation, it's still called here. I'd rename this to "expectedPvcPrime", and just validate that we get the expected result.

Reply:
Change from `expectedMutateObjects mutateObjects` to `expectedObjects vpObjects` to support all objects we want to verify after tests run